### PR TITLE
EV3-G API for Button

### PR DESCRIPTION
**Pull request by dwalton76**

### CODE REVIEW

**Comment by dwalton76** [comment]
For issue #354

**Comment by dlech** [comment]
Might I suggest using python-evdev to simplify things a bit here?

http://python-evdev.readthedocs.io/en/latest/

**Comment by dwalton76** [comment]
That looks handy for Buttons but what about Sensors?  I could just have any `_wait()` for a Sensor class drop into a `while True` loop and constantly check to see if anything changed and then someday if there is POLL support for Sensor clean it up.

**Comment by dlech** [comment]
For sensors, the only thing you can do for now is busy wait as you suggested.

File: debian/control [hunk 1]
```diff
@@ -3,7 +3,7 @@ Maintainer: Ralph Hempel <rhempel@hempeldesigngroup.com>
 Section: python
 Priority: optional
 Standards-Version: 3.9.5
-Build-Depends: python3-setuptools (>= 0.6b3), python3-all (>= 3.4), debhelper (>= 9), dh-python
+Build-Depends: python3-setuptools (>= 0.6b3), python3-all (>= 3.4), debhelper (>= 9), dh-python, python-evdev
```
Comment (dlech) [inline debian/control]: This is build deps. You don't need to add anything here.

Debian automatically figures out python package dependencies, so you don't need to add it in the binary package either.

(And this is the python2 version of the package)

File: ev3dev/core.py [hunk 2]
```diff
@@ -137,6 +138,22 @@ def matches(attribute, pattern):
             if all([matches(path + '/' + k, kwargs[k]) for k in kwargs]):
                 yield f
 
+
+def file_open(path):
```
Comment (dlech) [inline ev3dev/core.py]: Private?

File: debian/control (hunk 1 above)
Comment (dwalton76) [reply debian/control]: reverted

File: ev3dev/core.py (hunk 2 above)
Comment (dwalton76) [reply ev3dev/core.py]: Actually, I can undo this change, it is no longer needed.

**Comment by dwalton76** [comment]
How do I update the CI test to install python-evdev?

**Comment by dlech** [comment]
Like this?

https://github.com/rhempel/ev3dev-lang-python/blob/develop/.travis.yml#L6

**Comment by dlech** [comment]
Or you can change `sudo` to `true` and add an install line for `sudo apt-get install --yes python3-evdev`

(probably needs `sudo apt-get update` first)

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 language: python
 python:
 - 3.4
-sudo: false
+sudo: true
```
Comment (WasabiFan) [inline .travis.yml]: I'd prefer not to do this... a) we already use pip for the other dependency and b) this switches the builds from container-based to full VMs which they claim are slower.

**Comment by dwalton76** [comment]
It looks like it is installing but fails to import in api_tests.py....will dig more tomorrow

**Comment by ddemidov** [comment]
Sorry for being late to the train, but is there `python-evdev` for micropython?

**Comment by dwalton76** [comment]
I can't find where there is micropython support for python-evdev.  For micropython we could always drop into a loop instead of being event driven...not great but would be a work-around.

**Comment by dlech** [comment]
No, micropython does not support anything that is not in the standard library and only a fraction of the standard library. We can easily implement something very similar to the parts of evdev that we need in micropython, but not in a way that is compatible with regular python (e.g. you need to use uctypes which is not api compatible with ctypes).

**Comment by dwalton76** [comment]
I need to get this in to finish the work on #388 and I think this one is in pretty good shape (no comments recently) so I am going to go ahead and merge it.

**Pull request merged**